File: Java/199_Binary_Tree_Right_Side_View.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if(root==null){
            return new ArrayList<Integer>();
        }
        Deque<TreeNode> queue=new ArrayDeque<TreeNode>();
        queue.addLast(root);
        ArrayList<Integer> res=new ArrayList<Integer>();
        while(queue.peek()!=null){
            res.add(queue.getLast().val);
            Deque<TreeNode> cur=new ArrayDeque<TreeNode>();
            while(queue.peek()!=null){
                TreeNode p=queue.removeFirst();
                if(p.left!=null) cur.addLast(p.left);
                if(p.right!=null) cur.addLast(p.right);
            }
            queue=cur;
        }
        return res;
    }
}

File: python/199_Binary_Tree_Right_Side_View.py
# equals to find all most right node in breadth first search(level order traversal)

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {integer[]}
    def rightSideView(self, root):
        if not root:
            return []
        queue=[root]
        res=[]
        while queue:
            res.append(queue[-1].val)
            cur=[]
            for node in queue:
                if node.left:
                    cur.append(node.left)
                if node.right:
                    cur.append(node.right)
            queue=cur
        return res